palantir-java-format/src/main/java/com/palantir/javaformat/java/JavaInputAstVisitor.java

```diff
@@ -3304,13 +3305,22 @@ private Integer actualColumn(ExpressionTree expression) {
 
     /** How many lines does this node take up in the input. Returns at least 1. */
     int lineSpan(Tree node) {
-        IntFunction<Integer> lineNumberAt = tokenPosition -> {
-            Input.Token token = builder.getInput().getPositionTokenMap().get(tokenPosition);
-            return builder.getInput().getLineNumber(token.getTok().getPosition());
-        };
-        return lineNumberAt.apply(getEndPosition(node, getCurrentPath()))
-                - lineNumberAt.apply(getStartPosition(node))
-                + 1;
+        ImmutableRangeMap<Integer, ? extends Input.Token> positionTokenMap =
```

Comment:
```suggestion
        RangeMap<Integer, ? extends Input.Token> positionTokenMap =
```

Reply:
I thought immutables recommend using them as types directly

Reply:
Yeah I think there is actually an error-prone rule for this, because using the `Immutbale*` type ensures that if anyone further down tries to call one of the mutator methods (e.g. put) then they get a nice deprecation message.